File: src/main/java/org/nspectator/katas/FizzBuzz/FizzBuzz.java
package org.nspectator.katas.FizzBuzz;

public class FizzBuzz {

    public static String getResult(int number) {
        if (number % 15 == 0) return "fizzbuzz";
        else if (number % 3 == 0) return "fizz";
        else if (number % 5 == 0) return "buzz";
        return Integer.toString(number);
    }

}

File: src/main/scala/org/nspectator/fizzbuzz/FizzBuzz.scala
package org.nspectator.fizzbuzz

trait FizzBuzz {
  def getResult(number: Int): String
}

class FizzBuzz1 extends FizzBuzz {

  def getResult(number: Int) = {
    var result = ""
    if (number % 3 == 0) result = "fizz"
    if (number % 5 == 0) result += "buzz"
    if (result == "") number.toString
    else result
  }

}

class FizzBuzz2 extends FizzBuzz {

  def getResult(number: Int): String = (number % 3, number % 5) match {
    case (0 , 0) => "fizzbuzz"
    case (0 , _) => "fizz"
    case (_ , 0) => "buzz"
    case  _      => number.toString
  }

}